cmd/genji/shell/command.go

```diff
@@ -106,3 +107,70 @@ func runHelpCmd() error {
 
 	return nil
 }
+
+// executeInsertCommand insert json data.
+func executeInsertCommand(db *genji.DB, table string, data []byte) error {
+	q := fmt.Sprintf("INSERT INTO %s VALUES ?", table)
+	dec := json.NewDecoder(bytes.NewReader(data))
+	t, err := dec.Token()
+	if err != nil {
+		return err
+	}
+
+	d, ok := t.(json.Delim)
+	// check if the token is a json delimiter.
+	if !ok {
+		return fmt.Errorf("found %q, expected '{' or '['", d.String())
+	}
+
+	switch d {
+	case '[': // Array json stream
+		var m map[string]interface{}
+		for dec.More() {
+			err := dec.Decode(&m)
+			if err != nil {
+				return err
+			}
+
+		}
+
+		// Read the last token.
+		t, err := dec.Token()
+		if err != nil {
+			return fmt.Errorf("found %q, expected ']'", err.Error())
+		}
+
+		// Check the array is correctly closed.
+		if d, ok := t.(json.Delim); ok && d.String() != "]" {
+			return fmt.Errorf("found %q, expected ']'", d.String())
+		}
+
+		// Json format is valid.
+		// Convert map to a document.
+		d, err := document.NewFromMap(m)
```

Comment:
IIRC, you used [`document.NewFromJSON`](https://sourcegraph.com/github.com/genjidb/genji/-/blob/document/json.go#L16:6) at some point which was a good idea, much better than what I proposed initially, any reason why you are now using `NewFromMap` instead?